Gems/Atom/Feature/Common/Assets/Shaders/PostProcessing/EditorModeBlur.azsl

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <EditorModeCommon.azsli>
+#include <EditorModeDepthTransition.azsli>
+
+partial ShaderResourceGroup PassSrg : SRG_PerPass_WithFallback
+{
+    //! Width of kernal to apply box blur effect.
+    float m_kernalWidth;
+}
+
+//! Applies a box blur of the specified kernal width
+//! @note This is not an efficient implementation. Future work will replace this with a multi-pass Gaussian blur.
+float3 Blur(const float2 uv, const float kernalWidth)
+{
+    const float2 maskSize = PassSrg::GetImageSize(PassSrg::m_entityMask);
+    const float2 unormalizedUv = float2(uv.x * maskSize.x, uv.y * maskSize.y);
+    const float2 maskSizeReciprocal = float2(1., 1.0) / maskSize;
+    float3 outColor = float3(0., 0., 0.);
+
+    float weights = 0.;
+    for(float u = -kernalWidth; u <= kernalWidth; u+=1.)
+    {
+        for(float v = -kernalWidth; v <= kernalWidth; v+=1.)
+        {
+            weights += 1.;
+            outColor += PassSrg::m_framebuffer.Sample(PassSrg::PointSampler, (unormalizedUv + float2(u,v)) * maskSizeReciprocal).rgb;
+        }
+    }
+
+    return outColor / weights;
+}
+
+PSOutput MainPS(VSOutput IN)
+{
+    PSOutput OUT;
+
+    const float3 inColor = PassSrg::m_framebuffer.Sample(PassSrg::LinearSampler, IN.m_texCoord).rgb;
+    const float mask = PassSrg::m_entityMask.Sample(PassSrg::PointSampler, IN.m_texCoord);
```

Comment:
You're reading a float from a float4 texture, so best to specify which channel you're reading from, i.e. add a .r at the end of the sample:
const float mask = PassSrg::m_entityMask.Sample(PassSrg::PointSampler, IN.m_texCoord)**.r**;

Reply:
@antonmic The main issue is the final output is a new color render target. This would effect the connections of passes after this pass in the pass tree. And it's a risk for any passes added after this pass.

Reply:
Also, it's increase peak heap usage because the coexists of input color and new output color.

Reply:
Good catch, the mask will most likely end up being a single channel image format after `LYN-9878` but I'll get this fixed (and will check if there's any other instances of this) 👍